ros2run/ros2run/api/__init__.py

```diff
@@ -66,6 +67,13 @@ def run_executable(*, path, argv, prefix=None):
             # the subprocess will also receive the signal and should shut down
             # therefore we continue here until the process has finished
             pass
+    if process.returncode != 0:
+        if os.name == 'posix':
+            # a negative value -N indicates that the child was terminated by signal N.
+            print(signal.strsignal(-process.returncode))
```

Comment:
If the process doesn't crash, but returns a non-zero exit code, this ends up with:

```
ValueError: signal number out of range
```

So we'll have to handle that differently.

Reply:
Ah i see, nice catch 👍 thanks. i will do check valid signal range before that.